File: modia-syforest-provider/src/main/java/no/nav/sbl/dialogarena/modiasyforest/utils/DistinctFilter.java
package no.nav.sbl.dialogarena.modiasyforest.utils;


import java.util.HashSet;
import java.util.Set;
import java.util.function.Function;
import java.util.function.Predicate;

public class DistinctFilter<T, U> {
    private Set<U> set = new HashSet<>();

    public Predicate<T> on(Function<T, U> ext) {
        return t -> set.add(ext.apply(t));
    }
}

File: src/main/kotlin/no/nav/syfo/util/DistinctFilter.kt
package no.nav.syfo.util

import java.util.*
import java.util.function.Function
import java.util.function.Predicate

class DistinctFilter<T, U> {
    private val set: MutableSet<U> = HashSet()
    fun on(ext: Function<T, U>): Predicate<T> {
        return Predicate { t: T -> set.add(ext.apply(t)) }
    }
}
